app/src/NativeBase/Components/Forms/DatePicker.tsx

```diff
@@ -0,0 +1,195 @@
+import { VStack, Box, Text, Input, useColorMode, Icon } from 'native-base'
+import dayjs from 'dayjs'
+import StaTheme from '../../Theme/StaTheme'
+import MaterialIcons from 'react-native-vector-icons/MaterialIcons'
+
+interface DatePickerProps {
+  description: string
+  maximumDate?: Date
+  minimumDate?: Date
+  onChange: (date: Date) => void
+  value: Date
+}
+
+const DatePicker: FC<DatePickerProps> = ({
+  description,
+  maximumDate,
+  minimumDate,
+  onChange,
+  value,
+}) => {
+  const [pickerActive, setPickerActive] = useState(false)
+  const [selectedStartDate, setSelectedStartDate] = useState(null)
+  const [todaySelected, setTodaySelected] = useState(false)
+  const { colorMode } = useColorMode()
+  const { width } = Dimensions.get('window')
+
+  const handleChange = (value: Date) => {
+    onChange(value)
+    setSelectedStartDate(value)
+    setPickerActive(false)
+    if (dayjs(value).format('DD/MM/YY') === dayjs().format('DD/MM/YY')) {
+      setTodaySelected(true)
+    } else {
+      setTodaySelected(false)
+    }
+  }
+
+  const customDatesStylesCallback = (date: Date) => {
+    if (dayjs(date).format('DD/MM/YY') === dayjs().format('DD/MM/YY')) {
+      if (colorMode === 'light') {
+        return {
+          style: {
+            backgroundColor: 'bg.100',
+            borderWidth: 1,
+          },
+        }
+      } else {
+        return {
+          style: {
+            borderColor: 'white',
+            backgroundColor: 'bgDarkMode.100',
+            borderWidth: 1,
+          },
+        }
+      }
+    }
+  }
+
+  return (
+    <VStack>
+      <Box marginTop="5" marginBottom="10" borderColor="border.100" shadow="2">
+        <VStack>
+          <Box
+            marginTop="1"
+            marginX="0.5"
+            shadow="1"
+            bg={
+              !pickerActive && colorMode === 'light'
+                ? 'lightGrey.100'
+                : 'primary.80'
+            }
+          >
+            <Text
+              paddingTop="7"
+              fontSize="xs"
+              lineHeight="7"
+              color={
+                !pickerActive && colorMode === 'light'
+                  ? 'text.100'
+                  : 'textDarkMode.100'
+              }
+              paddingLeft="6"
+            >
+              {description}
+            </Text>
+          </Box>
+          <Text
+            paddingTop="8"
+            fontSize="sm"
+            fontWeight="600"
+            lineHeight="7"
+            paddingLeft="6"
+            color={
+              !pickerActive && colorMode === 'light'
+                ? 'text.100'
+                : 'primary.100'
+            }
+          >
+            Enter Date
+          </Text>
+          <Pressable onPress={() => setPickerActive(!pickerActive)}>
+            <Input
+              marginX="6"
+              value={dayjs(value).format('DD/MM/YY')}
+              accessibilityLabel="picker for available from date"
```

Comment:
Tiny thing, but if this is a generic date picker I wonder if the description here should be more general e.g. picker for date